pkg/operator/nodecadaemon_test.go

```
@@ -0,0 +1,207 @@
+				t.Fatalf("unexpected sync error: %v", err)
+			}
+
+			if tt.expectFailureSinceSet && controller.syncFailureSince.IsZero() {
+				t.Error("expected syncFailureSince to be set")
+			}
+			if !tt.expectFailureSinceSet && !controller.syncFailureSince.IsZero() {
+				t.Error("expected syncFailureSince to be reset")
+			}
+
+			cond := getNodeCACondition(t, regClient, "NodeCADaemonControllerDegraded")
+			switch tt.expectDegradedStatus {
```

golangci-lint (2.12.2)

[error] missing cases in switch of type v1.ConditionStatus: v1.ConditionTrue, v1.ConditionFalse, v1.ConditionUnknown

(exhaustive)